src/viewport/size/browser-viewport-size.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from "@angular/core";
+
+import { WindowRef } from "../../platform/window";
+import { IViewportSize, ViewportSize } from "./viewport-size.model";
+
+@Injectable({
+	providedIn: "root"
+})
+export class BrowserViewportSizeService implements IViewportSize {
```

Comment:
You can consolidate these so you have


- `ViewportSizeService` (you can simply use this as the Browser leave with providedIn root) 
- `ServerViewportSizeService extends ViewportSizeService` (remove providedIn root)
- Inject `ViewportSizeService` for usage
- Within Module add `forServer` which provide `ViewportSizeService` to use `ServerViewportSizeService `